src/app/calendar/calendar.cpp

```diff
@@ -6,7 +6,8 @@ void Calendar::createCalendar()
 {
     if (!Clinic::getCalendar())
     {
-        Clinic::appendCalendar(std::make_shared<Calendar>());
+        Calendar calendar;
+        Clinic::appendCalendar(std::make_shared<Calendar>(calendar));
```

Comment:
No need to create local calendar and than copying it to shared_ptr.
I didn't check the rest of the review, so I have no idea about the rest.

But it would be better to achieve the same just with:
```
Clinic::appendCalendar(std::make_shared<Calendar>());
```